### ISSUE TITLE
Remove Mac poll from cmake build

### ISSUE THREAD

**Issue by jay** [curl/curl#1089]
Ref #1057 

poll doesn't work right on some Mac versions so it's been disabled for all Mac in the autotools build. My understanding similar changes are needed for the cmake build. It was suggested I start a separate issue for that, so here's a c&p of my question in that thread:

---

for example should it be

``` cmake
if(NOT APPLE)
check_symbol_exists(poll          "${CURL_INCLUDES}" HAVE_POLL)
endif()
```

also should there be some sort of failsafe in setup like

```
#if defined(__APPLE__) && (defined(HAVE_POLL) || defined(HAVE_POLL_FINE))
#error "failsafe: we don't allow poll on apple, build config should have detected this"
#endif
```

this way it doesn't get away from us as more build systems are added or say for example someone adds HAVE_POLL_FINE to cmake

---

basically the idea is we want to make sure no poll on mac

@jzakrzewski @snikulov @Lekensteyn

**Comment by snikulov** [comment]
@jay Unfortunatelly, I have no Mac access, so I'm unable to help with this, sorry.
